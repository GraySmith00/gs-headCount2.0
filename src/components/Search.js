import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Search extends Component {
  constructor() {
    super();
    this.state = {
      searchValue: ''
    };
  }

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  render() {
    return (
      <div>
        <form>
          <input
            type="search"
            name="searchValue"
            onChange={this.handleChange}
            value={this.state.searchValue}
          />
        </form>
      </div>
    );
  }
}

export default Search;
