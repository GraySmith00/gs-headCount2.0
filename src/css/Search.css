input {
  padding: 10px 20px 10px 20px;

  font-size: 1.5em;
  outline: none;
  border: none;
  border-radius: 10px;
  opacity: 0.8;
}
