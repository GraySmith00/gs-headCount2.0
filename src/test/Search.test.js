import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Search from '../components/Search';

describe('Search component', () => {
  let wrapper;

  beforeEach(() => {
    shallow(<Search />);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it.skip('should change the searchValue state on input change', () => {});
});
