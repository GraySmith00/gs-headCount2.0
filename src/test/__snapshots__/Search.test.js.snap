// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should match the snapshot 1`] = `"<div><form><input type=\\"search\\" name=\\"searchValue\\" value=\\"\\" class=\\"search-input\\" placeholder=\\"Search for a district...\\"/></form></div>"`;
